000-ml-project-discovery.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: ml_project_discovery
+- Start Date: 2020-10-21
+- RFC PR: [amundsen-io/rfcs#000-ml-project-discovery.md](https://github.com/amundsen-io/rfcs/pull/0000) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# ML project discovery using Amundsen
+
+## Summary
+
+Our proposal is to extend the capabilities of Amundsen by introducing new data resources covering the ML project lifecycle.
+
+## Motivation
+
+The rising number of ML projects force organizations to address challenges related to model governance and model discovery.
+According to latest market surveys one can expect significant acceleration in AI adoption this and next year.
+
+## Guide-level Explanation (aka Product Details)
+
+The concept of machine learning project refers to complex process of moving your research from local exploration environment towards production grade deployments.
+Essentially, that consists of multiple steps including feature engineering, tracking experiments, model training, batch scoring, model versioning, model staging, release and deployment.
+Many organizations suffers from lack of visiblity over the ML assets delivered by the different teams of data scientists.
+Such model catalog would improve the collaboration within the organization through single interface to discover available ML resources and related metadata.  
+
+The basic use cases can be described as follow:
+
+*   search for ml projects by project name, project description, project owner
+*   search for ml models by model name, model description, model owner, model stage
+*   search for model deployments by deployment name, deployment description, deployment sample request/response (prediction)
+*   view the data sources used to build the model 
+*   view the metadata of the model, basic evaluation metrics e.g. accuracy, precision, recall, f1 etc.
+
+## UI/UX-level Explanation
```

Comment:
@dwarszawski do you or does ING have any designer which could help to drive certain UX mock for the model? that will be great

Reply:
Let's also collaborate on it either way, without dedicated design support this part might move slower but I can help make a best effort to align what is needed with existing design concepts.

Reply:
Apologize for late response. @feng-tao @ttannis I couldn't agree more. We should follow up on the UX part. Maybe next community meeting is a good time to start discussion?

Reply:
For the most part it'd be valuable to first have an understanding of all the generic kinds of metadata that apply to ML features, what is read-only vs editable etc. It'd be great to see the community discuss that first, and then consider the UI after that's settled.

Reply:
We are considering ml discovery and lineage as the essential parts of the ml project cycle. Different kind of metadata can be recognized at various stages of the project. From feature engineering (set of columns with description, statistics, normalization methods) through model training (number of iterations, batch size, optimizer, loss function etc) to model deployment (model owner, use case description,  stage of the model, performance reports).